flink-end-to-end-tests/test-scripts/test_streaming_file_sink.sh

```diff
@@ -45,47 +40,55 @@ function wait_for_restart {
 }
 
 ###################################
-# Wait a specific number of successful checkpoints
-# to have happened
+# Get all lines in part files and sort them numerically.
 #
 # Globals:
-#   None
+#   OUTPUT_PATH
 # Arguments:
-#   $1: the job id
-#   $2: the number of expected successful checkpoints
-#   $3: timeout in seconds
+#   None
 # Returns:
 #   None
 ###################################
-function wait_for_number_of_checkpoints {
-    local job_id=$1
-    local expected_num_checkpoints=$2
-    local timeout=$3
-    local count=0
-
-    echo "Starting to wait for completion of ${expected_num_checkpoints} checkpoints"
-    while (($(get_completed_number_of_checkpoints ${job_id}) < ${expected_num_checkpoints})); do
+function get_complete_result {
+    find "${OUTPUT_PATH}" -type f \( -iname "part-*" \) -exec cat {} + | sort -g
+}
 
-        if [[ ${count} -gt ${timeout} ]]; then
-            echo "A timeout occurred waiting for successful checkpoints"
+###################################
+# Waits until a number of values have been written within a timeout.
+# If the timeout expires, exit with return code 1.
+#
+# Globals:
+#   None
+# Arguments:
+#   $1: the number of expected values
+#   $2: timeout in seconds
+# Returns:
+#   None
+###################################
+function wait_for_complete_result {
+    local expected_number_of_values=$1
+    local polling_timeout=$2
+    local polling_interval=1
+    local seconds_elapsed=0
+
+    local number_of_values=$(get_complete_result | tail -1)
+    local previous_number_of_values=-1
+
+    while [[ ${number_of_values} -lt ${expected_number_of_values} ]]; do
+        if [[ ${seconds_elapsed} -ge ${polling_timeout} ]]; then
```

Comment:
Using wall-clock time has other drawbacks, such as DST clock shifts.

Reply:
I meant unix epoch time in seconds:
`date +%s`

Reply:
Sorry, I was mistaken; DST clock shifts are not a problem. However, changing the system time changes the value of `date +%s`, which can be problematic. It is unlikely that the system time is adjusted while the tests runs but not impossible. In the current solution we would only risk overcounting. Also, in our test suite this is a common pattern to measure elapsed time:
https://github.com/apache/flink/blob/f2b549fea8c831b60368796ac323109967b8062e/flink-end-to-end-tests/test-scripts/common.sh#L653-L660
https://github.com/apache/flink/blob/f2b549fea8c831b60368796ac323109967b8062e/flink-end-to-end-tests/test-scripts/test_queryable_state_restart_tm.sh#L146-L158

I'll leave it as it is.

Reply:
Ok, I see your point. My concern was that although here it does not really matter because `get_complete_result` is very quick but in general counting number of lines can take longer, e.g. in coming test with s3 as output (#6957). Then timeout will be prolonged. I believe, we can discuss it in #6957.